Docs: components updates (#5315)
langflow-ai/langflow: docs/docs/Components/components-vector-stores.md

---
title: Vector stores
slug: /components-vector-stores
---

# Vector store components in Langflow

Vector databases store vector data, which backs AI workloads like chatbots and Retrieval Augmented Generation.

Vector database components establish connections to existing vector databases or create in-memory vector stores for storing and retrieving vector data.

> Vector database components are distinct from [memory components](/components-memories), which are are built specifically for storing and retrieving chat messages from external databases.

Reviewer: ```suggestion
Vector database components are distinct from [memory components](/components-memories), which are built specifically for storing and retrieving chat messages from external databases.
```

## Use a vector store component in a flow

> Vector databases can be populated from within Langflow with document ingestion pipelines, like this one.

Reviewer: ```suggestion
Vector databases can be populated from within Langflow with document ingestion pipelines, like the following.
```

|-----------------------|--------------|------------------------------------------------------------|
| vectara_customer_id   | String       | Vectara customer ID                                        |
| vectara_corpus_id     | String       | Vectara corpus ID                                          |
| vectara_api_key       | SecretString | Vectara API key                                            |
| search_query          | String       | The query to receive an answer on                          |
| lexical_interpolation | Float        | Hybrid search factor (0.005 to 0.1)                        |
| filter                | String       | Metadata filters to narrow the search                      |
| reranker              | String       | Reranker type (mmr, rerank_multilingual_v1, none)          |
| reranker_k            | Integer      | Number of results to rerank (1 to 100)                     |
| diversity_bias        | Float        | Diversity bias for MMR reranker (0 to 1)                   |
| max_results           | Integer      | Maximum number of search results to summarize (1 to 100)   |

> | response_lang         | String       | Language code for the response (e.g., "eng", "auto")       |

Reviewer: Per the Google style guide, avoid "e.g.". Use "for example" instead.
